### PR TITLE
Adding a scheduled task which can be activated to remove non-approved/deleted questions completely.

### PR DESCRIPTION
**Pull request by Clyxz**
Fixes issue 255: Adding a scheduled task which can be activated to remove non-approved/deleted questions completely.
Currently non approved and deleted questions will be included for removal.

### CODE REVIEW

**Comment by dionysius** [comment]
The CI check failing part comes from master, fixed in https://github.com/frankkoch/moodle-mod_studentquiz/commit/72e62901097cb13baaf6fb711ac2fcf890bb14a5. No need to merge master again, just as a note here

**Review by dionysius — CHANGES_REQUESTED** [review]

File: classes/task/delete_orphaned_questions.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                    if (isset($question->questionid)) {
+
+                        $a = [
+                            'name' => format_string($question->name),
+                            'qtype' => format_string($question->qtype),
+                            'questionid' => format_string($question->questionid),
+                        ];
+
+                        $output .= get_string('deleteorphanedquestionsquestioninfo', 'mod_studentquiz', $a);
+
```
[the remark this answers is not in the document]
Comment (Clyxz) [reply classes/task/delete_orphaned_questions.php:87]: I added a try-catch transaction block around these parts.

File: settings.php [hunk 2]
```diff
@@ -140,4 +140,22 @@
...
+        '0'
+    ));
+
+    $choices = [
+        '15768000' => get_string('settingsdeleteorphanedtime6m', 'studentquiz'),
+        '31536000' => get_string('settingsdeleteorphanedtime1y', 'studentquiz'),
+        '63072000' => get_string('settingsdeleteorphanedtime2y', 'studentquiz'),
+        '94608000' => get_string('settingsdeleteorphanedtime3y', 'studentquiz')];
+
+    $settings->add(new admin_setting_configselect('studentquiz/deleteorphanedtimelimit',
```
[the remark this answers is not in the document]
Comment (Clyxz) [reply settings.php]: There is indeed a duration field for administration settings. I added this instead of the dropdown menu.

**Comment by dionysius** [comment]
Thank you very much for the contribution! :tada:

**Pull request merged**